### PR TITLE
Extract space-follows tests from integration-test

### PR DESCRIPTION
**Pull request by TarekkMA**
based on #74

### CODE REVIEW

File: pallets/space-follows/tests/src/mock.rs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        Block = Block,
+        NodeBlock = Block,
+        UncheckedExtrinsic = UncheckedExtrinsic,
+    {
+        System: frame_system,
+        Timestamp: pallet_timestamp,
+        Balances: pallet_balances,
+        Permissions: pallet_permissions,
+        Roles: pallet_roles,
+        Profiles: pallet_profiles,
```
Comment (F3Joule) [inline pallets/space-follows/tests/src/mock.rs:28]:
```suggestion
```

File: pallets/space-follows/tests/src/mock.rs [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        SpaceFollows: pallet_space_follows,
+        Posts: pallet_posts,
```
Comment (F3Joule) [inline pallets/space-follows/tests/src/mock.rs:30]:
```suggestion
```

File: pallets/space-follows/tests/src/mock.rs [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    pub const MaxCommentDepth: u32 = 10;
+}
+
+impl pallet_posts::Config for Test {
+    type Event = Event;
+    type MaxCommentDepth = MaxCommentDepth;
+    type AfterPostUpdated = ();
+    type IsPostBlocked = MockModeration;
+}
+
```
Comment (F3Joule) [inline pallets/space-follows/tests/src/mock.rs:110]:
```suggestion
```

File: pallets/space-follows/tests/src/mock.rs [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    pub const MaxUsersToProcessPerDeleteRole: u16 = 40;
+}
+
+impl pallet_roles::Config for Test {
+    type Event = Event;
+    type MaxUsersToProcessPerDeleteRole = MaxUsersToProcessPerDeleteRole;
+    type SpacePermissionsProvider = Spaces;
+    type SpaceFollows = SpaceFollows;
+    type IsAccountBlocked = MockModeration;
+    type IsContentBlocked = MockModeration;
```
Comment (F3Joule) [inline pallets/space-follows/tests/src/mock.rs:126]:
```suggestion
    type IsAccountBlocked = ();
    type IsContentBlocked = ();
```

File: pallets/space-follows/tests/src/mock.rs [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+impl pallet_profiles::Config for Test {
+    type Event = Event;
+    type SpacePermissionsProvider = Spaces;
+}
+
```
Comment (F3Joule) [inline pallets/space-follows/tests/src/mock.rs:133]:
```suggestion
```

File: pallets/space-follows/tests/src/mock.rs [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+
+impl pallet_spaces::Config for Test {
+    type Event = Event;
+    type Roles = Roles;
+    type SpaceFollows = SpaceFollows;
+    type BeforeSpaceCreated = SpaceFollows;
+    type AfterSpaceUpdated = ();
+    type IsAccountBlocked = MockModeration;
+    type IsContentBlocked = MockModeration;
```
Comment (F3Joule) [inline pallets/space-follows/tests/src/mock.rs:142]:
```suggestion
    type IsAccountBlocked = ();
    type IsContentBlocked = ();
```

File: pallets/space-follows/tests/src/tests.rs [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+    ExtBuilder::build_with_space().execute_with(|| {
+        assert_ok!(_default_follow_space()); // Follow SpaceId 1 by ACCOUNT2
+
+        assert_eq!(
+            SpaceFollows::spaces_followed_by_account(ACCOUNT2),
+            vec![SPACE1]
+        );
+        assert_eq!(
+            SpaceFollows::space_followers(SPACE1),
+            vec![ACCOUNT1, ACCOUNT2]
```
Comment (F3Joule) [inline pallets/space-follows/tests/src/tests.rs:23]: We don't auto-follow space by space owner anymore:

```suggestion
            vec![ACCOUNT2]
```

File: pallets/space-follows/tests/src/tests.rs [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+        );
+        assert_eq!(
+            SpaceFollows::space_followed_by_account((ACCOUNT2, SPACE1)),
+            true
+        );
```
Comment (F3Joule) [inline pallets/space-follows/tests/src/tests.rs:28]:
```suggestion
        assert!(SpaceFollows::space_followed_by_account((ACCOUNT2, SPACE1)));
```

File: pallets/space-follows/tests/src/tests.rs [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+#[test]
+fn follow_space_should_fail_when_trying_to_follow_hidden_space() {
+    ExtBuilder::build_with_space().execute_with(|| {
+        assert_ok!(_update_space(
+            None,
+            None,
+            Some(space_update(None, None, Some(true)))
+        ));
```
Comment (F3Joule) [inline pallets/space-follows/tests/src/tests.rs:61]:
```suggestion
        assert_ok!(_update_space(None, None, Some(space_update(None, Some(true)))));
```

File: pallets/space-follows/tests/src/tests.rs [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+
+#[test]
+fn unfollow_space_should_work() {
+    ExtBuilder::build_with_space().execute_with(|| {
+        assert_ok!(_default_follow_space());
+        // Follow SpaceId 1 by ACCOUNT2
+        assert_ok!(_default_unfollow_space());
+
+        assert!(SpaceFollows::spaces_followed_by_account(ACCOUNT2).is_empty());
+        assert_eq!(SpaceFollows::space_followers(SPACE1), vec![ACCOUNT1]);
```
Comment (F3Joule) [inline pallets/space-follows/tests/src/tests.rs:78]:
```suggestion
        assert!(SpaceFollows::space_followers(SPACE1).is_empty());
```

**Review by F3Joule — CHANGES_REQUESTED** [review]

**Review by siman — CHANGES_REQUESTED** [review]

File: pallets/space-follows/tests/src/tests_utils.rs [hunk 11]
```diff
@@ -0,0 +1,640 @@
+use frame_support::assert_ok;
```
[the remark this answers is not in the document]
Comment (TarekkMA) [reply pallets/space-follows/tests/src/tests_utils.rs:1]: I'm not refactoring any code, just copying from integration-test

**Review by F3Joule — APPROVED** [review]

**Review by F3Joule — APPROVED** [review]

File: pallets/space-follows/tests/Cargo.toml [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+description = 'Tests for Space Follows pallet'
+keywords = ['blockchain', 'cryptocurrency', 'social-network', 'news-feed', 'marketplace']
+categories = ['cryptography::cryptocurrencies']
+
+[package.metadata.docs.rs]
+targets = ["x86_64-unknown-linux-gnu"]
+
+[dependencies]
+codec = { package = "parity-scale-codec", version = "3.0.0", default-features = false, features = ["derive"] }
+scale-info = { version = "2.1.1", default-features = false, features = ["derive"] }
```
Comment (F3Joule) [inline pallets/space-follows/tests/Cargo.toml:18]:
```suggestion
scale-info = { version = "2.1.2", default-features = false, features = ["derive"] }
```

**Review by olehmell — APPROVED** [review]

File: pallets/space-follows/tests/src/tests_utils.rs [hunk 13]
```diff
@@ -0,0 +1,153 @@
...
+pub(crate) fn _create_space(
+    origin: Option<Origin>,
+    handle: Option<Option<Vec<u8>>>,
+    content: Option<Content>,
+    permissions: Option<Option<SpacePermissions>>,
+) -> DispatchResultWithPostInfo {
+    _create_space_with_parent_id(origin, handle, content, permissions)
+}
+
+pub(crate) fn _create_subspace(
```
Comment (siman) [inline pallets/space-follows/tests/src/tests_utils.rs:94]: We don't have subspace functionality

File: pallets/space-follows/tests/src/tests_utils.rs [hunk 14]
```diff
@@ -0,0 +1,153 @@
...
+    origin: Option<Origin>,
+    handle: Option<Option<Vec<u8>>>,
+    content: Option<Content>,
+    permissions: Option<Option<SpacePermissions>>,
+) -> DispatchResultWithPostInfo {
+    _create_space_with_parent_id(origin, handle, content, permissions)
+}
+
+pub(crate) fn _create_space_with_parent_id(
```
Comment (siman) [inline pallets/space-follows/tests/src/tests_utils.rs:103]: We don't have parent space functionality

**Review by siman — CHANGES_REQUESTED** [review]

**Review by siman — APPROVED** [review]

**Pull request merged**